core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright 2019 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.test.generator
+
+import com.pingcap.tikv.row.ObjectRowImpl
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.test.generator.DataType._
+
+import scala.collection.mutable
+import scala.util.Random
+
+object TestDataGenerator {
+
+  type TiRow = com.pingcap.tikv.row.Row
+
+  val bits = List(BIT)
+  val booleans = List(BOOLEAN)
+  val integers = List(TINYINT, SMALLINT, MEDIUMINT, INT, BIGINT)
+  val decimals = List(DECIMAL)
+  val doubles = List(FLOAT, DOUBLE)
+
+  val timestamps = List(TIMESTAMP, DATETIME)
+  val dates = List(DATE)
+  val durations = List(TIME)
+  val years = List(YEAR)
+
+  val texts = List(TEXT, TINYTEXT, MEDIUMTEXT, LONGTEXT)
+  val strings = List(VARCHAR, CHAR)
+  val binaries = List(VARBINARY, BINARY)
+  val bytes = List(BLOB, TINYBLOB, MEDIUMBLOB, LONGBLOB)
+  val enums = List(ENUM)
+  val sets = List(SET)
+
+  val jsons = List(JSON)
+
+  // M indicates the number of digits
+  val withM: List[ReflectedDataType] = bits ::: integers ::: decimals ::: doubles
+  // D indicates the number of digits after the decimal point
+  val withD: List[ReflectedDataType] = decimals ::: doubles
+  // U indicates whether the datatype is unsigned
+  val withU: List[ReflectedDataType] = integers ::: decimals ::: doubles
+  // F indicates the fraction of date/datetime format
+  val withF: List[ReflectedDataType] = timestamps ::: durations
+
+  val numeric: List[ReflectedDataType] = bits ::: booleans ::: integers ::: decimals ::: doubles
+  val dateAndDateTime: List[ReflectedDataType] = timestamps ::: dates ::: durations ::: years
+
+  val stringAndBinaries: List[ReflectedDataType] = strings ::: binaries
+  // TODO: support enum and set
```

Comment:
please create an GitHub issue to track this and also paste the issue link here.

Reply:
okay

Reply:
#946 @zhexuany

Reply:
I mean you need copy and paste the issue link right after this todo.

Reply:
okay.